pkg/resourcecollector/persistentvolumeclaim.go

```diff
@@ -101,11 +102,23 @@ func (r *ResourceCollector) preparePVCResourceForApply(
 		// In the case of storageClassMappings, we need to reset the
 		// storage class annotation and the provisioner annotation
 		var newSc string
+		var currentSc string
 		var exists bool
 		var provisioner string
-		if val, ok := pvc.Annotations[v1.BetaStorageClassAnnotation]; ok {
-			if newSc, exists = storageClassMappings[val]; exists && len(newSc) > 0 {
-				pvc.Annotations[v1.BetaStorageClassAnnotation] = newSc
+		// Get the existing storage class from the pvc spec
+		// It can be in BetaStorageClassAnnotation annotation or in the spec.
+		currentSc, err := utils.GetStorageClassNameForPVC(&pvc)
+		if err != nil {
+			return false, err
+		}
+		if len(currentSc) != 0 {
+			if newSc, exists = storageClassMappings[currentSc]; exists && len(newSc) > 0 {
+				if _, ok := pvc.Annotations[v1.BetaStorageClassAnnotation]; ok {
+					pvc.Annotations[v1.BetaStorageClassAnnotation] = newSc
```

Comment:
Lets check for the non-beta (v1) annotations as well.

Reply:
Looks like there isn't a non-beta annotation.

Reply:
Yes, we have only beta version of annotation only,